### PR TITLE
APIKIT-2022: Get all references does not work properly.

### PR DESCRIPTION
**Pull request by JuanAller**

### CODE REVIEW

File: parser-service/src/main/java/org/mule/parser/service/strategy/AMFParsingStrategy.java [hunk 1]
```diff
@@ -6,9 +6,11 @@
  */
 package org.mule.parser.service.strategy;
 
+import static java.util.Collections.*;
 import static org.mule.parser.service.strategy.ValidationReportHelper.errors;
```
Comment (pangelani) [inline parser-service/src/main/java/org/mule/parser/service/strategy/AMFParsingStrategy.java]: Replace with specific imports

File: parser-service/src/test/java/org/mule/parser/service/RamlParsingTestCase.java [hunk 2]
```diff
@@ -28,6 +35,8 @@
 @RunWith(Parameterized.class)
 public class RamlParsingTestCase {
```
Comment (pangelani) [inline parser-service/src/test/java/org/mule/parser/service/RamlParsingTestCase.java]: Merge this suite with the other (GetAllReferences)

Comment (pangelani) [inline parser-service/src/test/java/org/mule/parser/service/RamlParsingTestCase.java]: Merge this suite with the other (GetAllReferences)

File: raml-parser-interface/src/main/java/org/mule/apikit/common/ReferencesUtils.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+public class ReferencesUtils {
+
+  private static String OS = null;
+
+  private ReferencesUtils() {}
+
+  public static URI toURI(String path) {
+    URI uri;
+    if (isWindows()) {
+      uri = URI.create(path.replaceAll("\\s+","%20").replaceAll("\\\\", "/"));
```
Comment (pangelani) [inline raml-parser-interface/src/main/java/org/mule/apikit/common/ReferencesUtils.java]: Wrong regex: remove + char

File: parser-service/src/main/java/org/mule/parser/service/strategy/RamlParsingStrategy.java [hunk 4]
```diff
@@ -36,23 +42,37 @@ public ParseResult parse(ApiReference ref) {
     }
   }
 
+  private List<String> getReferences(ApiReference ref) {
+    ParseResult result = parsingStrategy.parse(ref);
```
Comment (pangelani) [inline parser-service/src/main/java/org/mule/parser/service/strategy/RamlParsingStrategy.java]: Default behavior is to run AMF in the first place, so at this point is likely that AMF parsing has already been done. Can we use that first parse to store refs? (Maybe having a global cache for references)

File: interface-impl-v2/src/main/java/org/mule/apikit/implv2/ParserWrapperV2.java [hunk 5]
```diff
@@ -94,6 +98,6 @@ public ApiValidationReport validate() {
 
   @Override
   public ApiSpecification parse() {
-    return ParserV2Utils.build(resourceLoader, ramlPath);
+    return ParserV2Utils.build(resourceLoader, ramlPath, references);
```
[the remark this answers is not in the document]
Comment (juanchib) [reply interface-impl-v2/src/main/java/org/mule/apikit/implv2/ParserWrapperV2.java]: I see that the findIncludes  has been removed and the remaining code is only used by ParaserWrapperV2 I think we could/should move the code there.

**Pull request merged**